### PR TITLE
Increase default service-controller parallelism

### PR DESCRIPTION
**Pull request by shyamjvs**
Based off of @MrHohn 's suggestion. Ref: https://github.com/kubernetes/kubernetes/issues/52495#issuecomment-337106218

I think we should make this the cluster default. As current situation is not very desirable, i.e multiple updates (potentially small) could be waiting for a long time because of a single big update (possibly due to external LB creation).

cc @kubernetes/sig-network-misc @nicksardo

### CODE REVIEW

**Comment by MrHohn** [comment]
@shyamjvs Thanks! I'd love to see this change. Should we wait for new results from correctness CIs (ref https://github.com/kubernetes/test-infra/pull/5038)?

**Comment by shyamjvs** [comment]
This change should be useful anyway.. but sure, we can wait for it.

**Comment by shyamjvs** [comment]
Seems like increasing parallelism was useful - https://github.com/kubernetes/kubernetes/issues/52495#issuecomment-339693141

@MrHohn Should we let this change in now?

**Review by MrHohn — APPROVED** [review]
Thanks for the update, this LGTM :)
/lgtm

**Comment by MrHohn** [comment]
Might need approval from @thockin.
/assign @thockin

**Comment by nicksardo** [comment]
I'm not against raising the concurrency, but it does make me wonder whether all the cloud providers (ourselves included) can handle the concurrency. Should this be brought up at sig-network or wg-cloudprovider? Even if we increase the concurrency for GCE, I'd say we need more testing/confidence. 

@MrHohn just told me that the concurrency doesn't affect the UpdateLoadBalancer call which adds/removes instances to target pools. That's unfortunate considering that some users with a significant amount of load balancers say that target pool management is too slow. https://github.com/kubernetes/kubernetes/issues/39423

**Comment by MrHohn** [comment]
> @mrhohn just told me that the concurrency doesn't affect the UpdateLoadBalancer call which adds/removes instances to target pools

Ref:
https://github.com/kubernetes/kubernetes/blob/8f505e4c8602673496916b9363e74b66b6f90c7c/pkg/controller/service/service_controller.go#L665-L711

**Comment by shyamjvs** [comment]
I agree that we should discuss potential cloud-provider specific consequences this might have.

> @mrhohn just told me that the concurrency doesn't affect the UpdateLoadBalancer call which adds/removes instances to target pools.

Indeed quite surprising :)

**Comment by MrHohn** [comment]
Adding hold as we haven't taken any action w.r.t. https://github.com/kubernetes/kubernetes/pull/54057#issuecomment-339735128 yet.

/hold

**Comment by shyamjvs** [comment]
Ping - has there been some decision here?

**Comment by shyamjvs** [comment]
Closing this PR as it was agreed this doesn't offer much advantage for now. We can reopen it later if needed.

**Pull request closed without merging**
